python/src/trezorlib/btc.py

```diff
@@ -17,17 +17,57 @@
 import warnings
 from copy import copy
 from decimal import Decimal
-from typing import TYPE_CHECKING, Any, Dict, Sequence, Tuple
+from typing import TYPE_CHECKING, Any, AnyStr, Dict, List, Optional, Sequence, Tuple
+
+# TypedDict is not available in typing for python < 3.8
+from typing_extensions import TypedDict
 
 from . import exceptions, messages
 from .tools import expect, normalize_nfc, session
 
 if TYPE_CHECKING:
     from .client import TrezorClient
-
-
-def from_json(json_dict):
-    def make_input(vin):
+    from .tools import Address
+    from .protobuf import MessageType
+
+    class ScriptSig(TypedDict):
+        asm: str
+        hex: str
+
+    class ScriptPubKey(TypedDict):
+        asm: str
+        hex: str
+        type: str
+        reqSigs: int
+        addresses: List[str]
+
+    class Vin(TypedDict):
+        txid: str
+        vout: int
+        sequence: int
+        coinbase: str
```

Comment:
this should be an optional field: `NotRequired[str]`

Reply:
I am not able to get the `NotRequired` operator from anywhere (`typing`, `typing_extensions`, `from __future__ import annotations`). It looks like this is a fresh thing and is not in production yet - they say `Python-Version: | 3.11` at https://www.python.org/dev/peps/pep-0655/